art/estimators/speech_recognition/pytorch_icefall.py

```diff
@@ -0,0 +1,369 @@
+# MIT License
+#
+# Copyright (C) The Adversarial Robustness Toolbox (ART) Authors 2021
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+This module implements the task specific estimator for Icefall, an end-to-end speech recognition toolkit based on
+k2-fsa.
+
+| Repository link: https://github.com/k2-fsa/icefall/tree/master
+"""
+import ast
+from argparse import Namespace
+import logging
+from typing import Dict, List, Optional, Tuple, TYPE_CHECKING, Union
+
+import numpy as np
+import torch
+
+from art import config
+from art.estimators.pytorch import PyTorchEstimator
+from art.estimators.speech_recognition.speech_recognizer import SpeechRecognizerMixin, PytorchSpeechRecognizerMixin
+from art.utils import get_file
+
+if TYPE_CHECKING:
+    # pylint: disable=C0412
+
+    from art.defences.preprocessor.preprocessor import Preprocessor
+    from art.defences.postprocessor.postprocessor import Postprocessor
+    from art.utils import CLIP_VALUES_TYPE, PREPROCESSING_TYPE
+
+logger = logging.getLogger(__name__)
+
+
+class PyTorchIcefall(PytorchSpeechRecognizerMixin, SpeechRecognizerMixin, PyTorchEstimator):
+    """
+    This class implements a model-specific automatic speech recognizer using the end-to-end speech recognizer in
+    Icefall.
+
+    | Repository link: https://github.com/k2-fsa/icefall/tree/master
+    """
+
+    from icefall.utils import AttributeDict
+    from pathlib import Path
+    import k2
+
+    estimator_params = PyTorchEstimator.estimator_params + ["icefall_config_filepath"]
```

Comment:
```suggestion
    estimator_params = PyTorchEstimator.estimator_params
```
The `icefall_config_filepath` parameter is not longer in the constructor.